server/src/main/java/keywhiz/service/resources/automation/v2/SecretResource.java

```diff
@@ -378,6 +384,18 @@ public boolean backfillExpiration(@Auth AutomationClient automationClient, @Path
     }
 
     if (expiry != null) {
+      if (existingExpiry.isPresent()) {
+        long offset = existingExpiry.get().until(expiry, HOURS);
+        if (offset > 24 || offset < -24) {
+          logger.warn(
+              "Extracted expiration of secret {} differs from actual by more than {} hours.",
```

Comment:
Maybe log the extracted expiration, so that we have the option to manually reset the secret's expiration to match the certificate's expiration?